src/entity_extraction/entity_extraction_evaluation.py

```diff
@@ -56,7 +102,116 @@ def get_token_labels(labelled_entities, raw_text):
         else:
             token_labels[token_start] = f"B-{label}"
 
-    return token_labels
+    return split_text, token_labels
+
+def generate_confusion_matrix(
+    labelled_tokens: list,
+    predicted_tokens: list,
+    output_path: str,
+    model_name: str
+):
+    """
+    Generates confusion matrix 
+    
+    Parameters
+    ----------
+    labelled_tokens : list[list[str]]
+        The true labels per token.
+    predicted_tokens : list[list[str]]
+        The predicted labels per token.
+    output_path: str
+        Path to output the confusion matrix diagram
+    model_name: str
+        Name of the model to include in the diagram title
+    """
+    
+    def get_label_to_index(labels):
```

Comment:
nitpick: I don't think this should be a nested function since it's only used once